frontends/web/src/components/TaskOwnerPageComponents/Contexts.js

```diff
@@ -4,11 +4,40 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
 import { Container, Row, Form, Col, Card, Button } from "react-bootstrap";
 import DragAndDrop from "../DragAndDrop/DragAndDrop";
+import UserContext from "../../containers/UserContext";
 
 const Contexts = (props) => {
+  async function fetchTags(tid, rid) {
+    let result = await context.api.getAllTaskTags(tid, rid);
+    return result;
+  }
+
+  async function fetchSelectedTags(tid, rid) {
+    let result = await context.api.getAllRoundTags(tid, rid);
+    return result;
+  }
+
+  useEffect(() => {
+    fetchTags(props.tid, props.values.rid).then((result) => {
+      setTags(result);
+    });
+
+    fetchSelectedTags(props.tid, props.values.rid).then((result) => {
+      if (result) {
+        setSelectedTags(result);
+      } else {
+        setSelectedTags([]);
+      }
+    });
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const context = useContext(UserContext);
+  const [tags, setTags] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
```

Comment:
instead of having `selectedTags` as an array to store selection data, maybe consider using a `tagStatus` state that is a JS object where keys are the `tag` and values (bool) denote whether they've been selected, ala:

```js
{
 tag1: false,
 tag2: true,
 tag3: false,
 etc...
}
```

main motivation is that currently, every `onChange` (line 107) call results in a linear (O(N)) time `.filter` if it's unchecked, and a linear memory `newTags` if it's checked.

with `tags` being an object, onChange can now just be:

```js
onChange={(event) => {
 setTagStatus({ event.target.value : event.target.checked}); // constant time operation
}
```

How I'd handle the conversion of `round.selectedTags` to the hashmap is to initialise it in `Contexts.js` and then convert it back in `TaskOwnerPage`'s `handleRoundUpdate` so that it's a one-time computation.  

in Contexts.js's `useEffect`:
```js
fetchTags(props.tid, props.values.rid).then((result) => {
      setTags(result);
});

let initStatus = tags.reduce((a, v) => ({ ...a, [v]: false}), {})); // creates object where all tags set to false
let selectedTagObj = selectedTags.reduce((a, v) => ({ ...a, [v]: true}), {}));
Object.assign(initStatus, selectedTagObj); // sets selected tags to true
setTagStatus(initStatus)
```

then in TaskOwnerPage's `handleRoundUpdate`, just filter keys and convert to an array